lib/ace/ext/papeeria/spellchecker.coffee

```diff
@@ -37,196 +35,206 @@ define((require, exports, module) ->
-        editor.keyBinding.removeKeyboardHandler(@keyboard)
-
-        editor.off("change", @changeListener)
-        editor.off("changeSelection", @changeSelectionListener)
-        editor.off("blur", @blurListener)
-        editor.off("mousedown", @mousedownListener)
-        editor.off("mousewheel", @mousewheelListener)
-
-        @activated = false
-        return
-
-      @commands = {
-        "Esc": =>
-          @detach()
-        "Up": =>
-          @goTo("up")
-        "Down": =>
-          @goTo("down")
-        "Return": =>
-          @insertCorrection()
-      }
-
-      @changeListener = (e) =>
-        @detach()
-        return
-
-      @changeSelectionListener = (e) =>
-        cursor = editor.selection.lead
-        if cursor.row isnt @base.row or cursor.column < @base.column
-          @detach()
-        if not @activated
-          @detach()
-        return
-
-      @blurListener = (e) =>
-        element = document.activeElement
-        text = editor.textInput.getElement()
-        container = @popup and @popup.container
-        if element isnt text and element.parentNode isnt container and e.relatedTarget isnt text
-          @detach()
-        return
-
-      @mousedownListener = (e) =>
-        @detach()
-        return
-
-      @mousewheelListener = (e) =>
-        @detach()
-        return
-
-      @insertCorrection = () =>
-        @detach()
-        row = @popup.getRow()
-        correction = options[row].value
-        wordRange = getCurrentWordRange()
-        word = editor.session.getTextRange(wordRange)
-        if ' ' in word
-          wordRange.end.column--
-        editor.session.replace(wordRange, correction)
-        return
-
-      @init()
-      return
 
-    # Convert an array of string to popup eligible structure.
-    # @param {Array} corrections: array of strings with substitution options.
-    # @returns {Array}: array of jsons, actually.
-    convertCorrectionList = (corrections) ->
-      options = []
-      options.push({caption: item, value: item}) for item in corrections
-      return options
-
-    # Returns Range object that describes current word position.
-    # @param {Range}: current word range.
-    getCurrentWordRange = ->
-      session = editor.session
-      row = editor.getCursorPosition().row
-      col = editor.getCursorPosition().column
-      return session.getAWordRange(row, col)
-
-    # Get the word under the cursor.
-    # @returns {String}
-    extractWord = ->
-      session = editor.session
-      row = editor.getCursorPosition().row
-      wordRange = getCurrentWordRange()
-      start = wordRange.start.column
-      end = wordRange.end.column
-      # getAWordRange returns start and end positions with a trailing
-      # whitespace at the end (if there's a one), that's why replace is used
-      return session.getTextRange(wordRange).replace(/\s\s*$/, '')
+  # Sets up spellchecker popup routine and implements
+  exports.setupSpellCheckerPopup = (editor) ->
```

Comment:
Please define functions the normal way, without assignments to exports, and assign to exports at the module end. This will clearly expose what this module exports.

Or even better, refactor the way of working with requirejs, like @Mikhail-M did in [this commit](https://github.com/bardsoftware/ace/pull/70/commits/98fac5c8b48202c068e6a449e54053131012a44a)